enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,509 @@
+  `minimal`.
+- As a developer, I want a supported way to collect a subset of the metrics
+  exported by my operator and operands, while still collecting necessary metrics
+  for alerts, visualization of key indicators and Telemetry.
+- As a developer of a component (that does not yet implement a profile), I want
+  to extract metrics needed to implement said profile, based on the assets I
+  provide, or the ones gathered from the cluster based on a group of target
+  selectors, and a plug-in relabel configuration to apply within the monitor.
+- As a developer of a component (that does not, or only partially implements a
+  profile), I want to get information about any monitors that are not yet
+  implemented for any of the supported profiles that are offered.
+- As a developer of a component (that implements a profile), I want to verify if
+  all the profile metrics are present in the cluster, and which of the profile
+  monitors are affected if not. Also, I want additional information to narrow
+  down where these metrics are exactly being used.
+
+### Goals
+
+- Give users a supported and documented method to pick the amount of metrics
+  being collected by the platform, in a way that allows
+  cluster-monitoring-operator and other operators to provide their own
+  defaulting.
+- Other components of the cluster (alerts, console, HPA and VPA and telemetry)
+  should not be negatively affected by this feature.
+- Developers may optionally adhere to this feature, those that do not adhere
+  should not be impacted.
+
+### Non-Goals
+
+- Dynamically adjusting metrics scraped at runtime based on heuristics.
+- Allowing users to explicitly set a list or a regex of metrics that they would
+want to be included in a given profile.
+- Support the addition of profiles other than `full` and `minimal`.
+
+## Proposal
+
+This enhancement leverages label selectors to allow cluster-monitoring-operator
+to select the appropriate monitors for a given metrics collection profile. 
+
+Given this, the following steps would be necessary to implement this
+enhancement:
+
+- expose a configuration option in the cluster-monitoring-operator ConfigMap
+  that allows selecting the metrics collection profile;
+- OCP components that would like to implement metrics collection profiles would
+  have to provide at least 1 monitor for each supported metrics collection
+  profile.
+
+### Workflow Description
+
+The cluster-monitoring-operator (CMO) will be extended with a new field,
+`collectionProfile` under `prometheusK8s`. This new field will then allow users
+to configure their desired metrics collection profile while preserving the
+platform functionality (alerts, console, HPA and VPA and telemetry).
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      collectionProfile: full
```

Comment:
Enum values should be PascalCase

Reply:
I see. I'll send out a patch to support camel and pascal cases (instead of just the former), and deprecate it in a later release.

Also, for future references, is there a guide that outlines such practices that we follow across OpenShift?

Reply:
This is a general k8s API convention.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants